src/setup.ts

```diff
@@ -0,0 +1,47 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+import {validateCheckSum, getNotationCheckSum} from './lib/checksum';
+import { getNotationDownloadURL } from './lib/install';
+
+// setup sets up the Notation CLI.
+async function setup(): Promise<void> {
+    try {
+        // inputs from user
+        const version: string = core.getInput('version');
+        const notation_url: string = core.getInput('url');
+        const notation_checksum = core.getInput('checksum').toLowerCase();
+
+        // sanity check
+        if (!(notation_url || version)) {
+            throw new Error("user needs to provide either version of official Notation CLI or url of customized Notation CLI release")
+        }
```

Comment:
We can have a default version in another PR.

Reply:
Actually, this is a bug. We are already using a default version defined in setup/action.yml. So `version` here will always be non-empty. Updated the logic.